common/lib/dependabot/config/ignore_condition.rb

```diff
@@ -50,10 +50,10 @@ def ignore_patch(version)
         parts = version.split(".")
         return [] unless parts.size > 2
 
-        lower_parts = parts.first(2) + ["a"]
+        lower_parts = parts.first(3)
         upper_parts = parts.first(2)
         upper_parts[1] = upper_parts[1].to_i + 1
-        lower_bound = ">= #{lower_parts.join('.')}"
+        lower_bound = "> #{lower_parts.join('.')}"
         upper_bound = "< #{upper_parts.join('.')}"
         ["#{lower_bound}, #{upper_bound}"]
       end
```

Comment:
Nice catch on the current version being ignored 👍 

I'm wondering if we also want to be able to handle longer and shorter versions here, we currently bail if the version has less than 3 parts which prevents us creating patch ignores for `1.2`, which we could assume means `1.2.0` (we do this in api when creating ignore conditions from PR commands).

Something like this might fix the current bug and work for `1.2` and `1.2.3.1`:

```ruby
def ignore_patch(version)
    parts = version.split(".")
    version_parts = parts.fill(0, parts.length...4)
    lower_parts = version_parts.first(3) + [version_parts.last.to_i + 1] + ["a"]
    upper_parts = version_parts.first(1) + [version_parts[1].to_i + 1]
    lower_bound = ">= #{lower_parts.join('.')}"
    upper_bound = "< #{upper_parts.join('.')}"
    ["#{lower_bound}, #{upper_bound}"]
  end
```

But might need some special handling for non-numeric versions, e.g. `Finchley.SR3`

Reply:
I went down a bit of a yak shave here while trying to get my head around some test cases: https://github.com/dependabot/dependabot-core/pull/3593

Do these test cases seem legit to you?

cc @mctofu and @thepwagner